File: 1691-84-1352-product-of-the-last-k-numbers/1691-84-1352-product-of-the-last-k-numbers.java
import java.util.*;

class ProductOfNumbers {
    private List<Integer> v;

    public ProductOfNumbers() {
        v = new ArrayList<>();
        v.add(1);
    }

    public void add(int num) {
        if (num == 0) {
            v.clear();
            v.add(1);
        } else {
            v.add(v.get(v.size() - 1) * num);
        }
    }

    public int getProduct(int k) {
        return v.size() <= k ? 0 : v.get(v.size() - 1) / v.get(v.size() - 1 - k);
    }
}

/**
 * Example Usage:
 * ProductOfNumbers obj = new ProductOfNumbers();
 * obj.add(num);
 * int param_2 = obj.getProduct(k);
 */

File: 1691-84-1352-product-of-the-last-k-numbers/1691-84-1352-product-of-the-last-k-numbers.py
class ProductOfNumbers:
    def __init__(self):
        self.v = [1]

    def add(self, num: int) -> None:
        if num == 0:
            self.v = [1]
        else:
            self.v.append(self.v[-1] * num)

    def getProduct(self, k: int) -> int:
        return 0 if len(self.v) <= k else self.v[-1] // self.v[-1 - k]

# Example Usage:
# obj = ProductOfNumbers()
# obj.add(num)
# param_2 = obj.getProduct(k)
